compiler/rustc_typeck/src/check/cast.rs

```diff
@@ -561,26 +561,32 @@ impl<'a, 'tcx> CastCheck<'tcx> {
     }
 
     /// invald-ptr-to-int-cast lint: when casting a pointer to a numeric type it only makes sense
-    /// to cast to `usize`. Other cases are often bugs. See #81686 as an example.
+    /// to cast to `usize` or `u64`. Other cases are often bugs. See #81686 as an example.
     fn invalid_ptr_to_int_cast_lint(&self, fcx: &FnCtxt<'a, 'tcx>) {
         if self.expr_ty.is_fn() || self.expr_ty.is_raw_ptr() {
-            let is_integral_cast = matches!(self.cast_ty.kind(), ty::Int(_) | ty::Uint(_));
-            if is_integral_cast && !matches!(self.cast_ty.kind(), ty::Uint(ty::UintTy::Usize)) {
+            let cast_ty_kind = self.cast_ty.kind();
+            let is_integral_cast = matches!(cast_ty_kind, ty::Int(_) | ty::Uint(_));
+            if is_integral_cast
+                && !matches!(cast_ty_kind, ty::Uint(ty::UintTy::Usize | ty::UintTy::U64))
+            {
                 fcx.tcx.struct_span_lint_hir(
                     lint::builtin::INVALID_PTR_TO_INT_CAST,
                     self.full_expr.hir_id,
                     self.full_expr.span,
                     |lint| {
                         let msg = format!("casting pointer to `{}`", self.cast_ty);
                         let mut lint = lint.build(&msg);
-                        lint.help("pointers should only be cast to `usize`");
+                        lint.help("pointers should only be cast to `usize` or `u64`");
 
                         if let Ok(e_snippet) =
                             fcx.tcx.sess.source_map().span_to_snippet(self.expr.span)
                         {
                             lint.span_suggestion(
                                 self.full_expr.span,
-                                &format!("to cast to `{}`, cast to `usize` first", self.cast_ty),
+                                &format!(
+                                    "to cast to `{}`, cast to `usize` or `u64` first",
```

Comment:
Likewise, here, I think it's fine to just say "cast to `usize` first". We don't need to give two different recommendations.